### PR TITLE
Task/add firebase crashlytics logs

### PR DESCRIPTION
**Pull request by dakshsilence**

### CODE REVIEW

File: lib/screens/backup/backup_destination_screen.dart [hunk 1]
```diff
@@ -163,7 +167,9 @@ class BackupDestinationWidget extends StatelessWidget {
       );
     } catch (error) {
       print('Cannot verify backup: $error');
-      analyticManager.trackVerifyBackup(success: false, timeSinceVerify: cloudMessage(check.date), source: PageSource.get_started, error: error.toString());
+      FirebaseCrashlytics.instance.log('Cannot verify backup: $error');
```
Comment (NoahJinnn) [inline lib/screens/backup/backup_destination_screen.dart]: We need to catch CredentialException on this line. And to be careful, we should catch CredentialException in any API that uses `credential_manager` pub in `android_secure_storage.dart`

File: lib/screens/backup/backup_destination_screen.dart [hunk 2]
```diff
@@ -163,7 +167,9 @@ class BackupDestinationWidget extends StatelessWidget {
       );
     } catch (error) {
       print('Cannot verify backup: $error');
```
Comment (NoahJinnn) [inline lib/screens/backup/backup_destination_screen.dart]: Remove unused `print`

File: lib/screens/pairing/pair_screen.dart [hunk 3]
```diff
@@ -107,6 +114,9 @@ class _PairState extends State<PairScreen> {
   }
 
   void _showError(Object error, BackupSource source) {
+    FirebaseCrashlytics.instance.log(
+        'Error recovering from credentionals: $error ${error is CredentialException ? 'CredentialException Code : ${error.code}, CredentialException Messsage: ${error.message}, ${error.details}' : ''}');
```
Comment (NoahJinnn) [inline lib/screens/pairing/pair_screen.dart]: This condition check repeats many times, we should create a utility function to make code less verbose

File: lib/screens/backup/backup_destination_screen.dart (hunk 2 above)
Comment (dakshsilence) [reply lib/screens/backup/backup_destination_screen.dart]: These were already there. I think added by Vasyl. Should I remove all the print statements in the code

**Review by noahsilencelab — APPROVED** [review]
LGTM

**Pull request merged**